### PR TITLE
gis: update four examples to use solara viz

### PR DESCRIPTION
**Pull request by wang-boyu**
Partly addresses https://github.com/projectmesa/mesa-examples/issues/154.

Update the following four gis examples to use the new solara visualization:

- agents_and_networks
- geo_schelling
- geo_schelling_points
- geo_sir

Update their dependency of mesa-geo to v0.9.0a0.

Also update the folder structure of the geo_sir model from

```bash
geo_sir
├── agents.py
├── model.py
├── app.py
├── ...
```

to

```bash
geo_sir
├── geo_sir
│   ├── __init__.py
│   ├── agents.py
│   └── model.py
├── app.py
├── ...
```

to be consistent with other gis examples.

### CODE REVIEW

**Comment by EwoutH** [comment]
Awesome work!

If others don’t get to it, I will review Monday.

File: gis/geo_schelling/README.md [hunk 1]
```diff
@@ -16,10 +16,10 @@ NUTS-2 regions are the GeoAgents. The neighbors of a polygon are considered thos
...
-To run the model interactively, run `mesa runserver` in this directory. e.g.
+To run the model interactively, run `solara run app.py` in this directory. e.g.
 
 ```bash
-mesa runserver
+solara run app.py
 ```
 
-Then open your browser to [http://127.0.0.1:8521/](http://127.0.0.1:8521/) and press `Start`.
+Then open your browser to [http://127.0.0.1:8765/](http://127.0.0.1:8765/) and press `Start`.
```
Comment (rht) [inline gis/geo_schelling/README.md:25]: There is no "Start" button anymore. It is "▶".

Comment (wang-boyu) [reply gis/geo_schelling/README.md]: You're right! Fixed now.

**Comment by wang-boyu** [comment]
Also updated the agents_networks example model. Won't update any other examples in this PR.

**Review by rht — APPROVED** [review]

File: gis/geo_sir/geo_sir/model.py
Comment (EwoutH) [inline gis/geo_sir/geo_sir/model.py:1]: I would do it the other way around, and keep everything in a `gis/geo_sir`.

**Review by EwoutH — APPROVED** [review]
Thanks for working on this, this looks really nice!

Also great that you cleaned some old files up.

Go ahead and merge!

**Comment by EwoutH** [comment]
Personally, I would highly prefer adoption this structure for all examples. It's by far the most straightforward and clearly arranged one.

> ```shell
> geo_sir
> ├── agents.py
> ├── model.py
> ├── app.py
> ```

**Comment by wang-boyu** [comment]
I had some difficulties in using the flat folder structure. If `model.py` and `agents.py` are inside `gis/geo_sir` together with `app.py`, then `solara run app.py` gives the following errors:

```bash
    from .agents import PersonAgent
ImportError: attempted relative import with no known parent package
```

```bash
   from .agents import NeighbourhoodAgent, PersonAgent
ImportError: attempted relative import with no known parent package
```

I will then have to change all the imports from `.agent` to `agents`, for example:

```python
# from
from .agents import NeighbourhoodAgent, PersonAgent

# to
from agents import NeighbourhoodAgent, PersonAgent
```

Then `solara run app.py` runs fine. However, tests cases will throw errors:

```bash
___________________________________________________ ERROR collecting test_gis_examples.py ___________________________________________________
ImportError while importing test module '/Users/boyu/GitHubProjects/mesa-examples/test_gis_examples.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/Cellar/python@3.12/3.12.7/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_gis_examples.py:30: in <module>
    @pytest.mark.parametrize("model_class", get_models("gis"))
test_gis_examples.py:17: in get_models
    module = importlib.import_module(module_name)
/usr/local/Cellar/python@3.12/3.12.7/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
gis/geo_sir/model.py:7: in <module>
    from agents import NeighbourhoodAgent, PersonAgent
E   ModuleNotFoundError: No module named 'agents'
========================================================== short test summary info ==========================================================
ERROR test_gis_examples.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

So I simply use the updated folder structure that works for both `solara` and tests.

**Comment by wang-boyu** [comment]
With the approvals I'll go ahead and merge this, so that users can have a working frontent for these examples. Thanks for looking into this @rht @EwoutH.

**Comment by EwoutH** [comment]
Awesome! I would be totally fine with you merging faster/earlier, if you feel that would speed up your development of these models.

**Pull request merged**